usaspending_api/reporting/tests/integration/test_populate_reporting_agency_missing_tas.py

```diff
@@ -0,0 +1,128 @@
+        mommy.make(
+            "submissions.SubmissionAttributes", submission_id=2, reporting_fiscal_year=2019, reporting_fiscal_period=4
+        ),
+    ]
+    agency = mommy.make("references.ToptierAgency", toptier_agency_id=1, toptier_code="123")
+
+    treas_accounts = [
+        mommy.make(
+            "accounts.TreasuryAppropriationAccount",
+            treasury_account_identifier=1,
+            funding_toptier_agency_id=agency.toptier_agency_id,
+            tas_rendering_label="tas-1",
+        ),
+        mommy.make(
+            "accounts.TreasuryAppropriationAccount",
+            treasury_account_identifier=2,
+            funding_toptier_agency_id=agency.toptier_agency_id,
+            tas_rendering_label="tas-2",
+        ),
+    ]
+
+    approps = [
+        {"sub_id": sub[0].submission_id, "treasury_account": treas_accounts[0]},
+        {"sub_id": sub[0].submission_id, "treasury_account": treas_accounts[1]},
+        {"sub_id": sub[1].submission_id, "treasury_account": treas_accounts[1]},
+    ]
+    for approp in approps:
+        mommy.make(
+            "accounts.AppropriationAccountBalances",
+            submission_id=approp["sub_id"],
+            treasury_account_identifier=approp["treasury_account"],
+        )
+
+    gtas_rows = [
+        {
+            "treasury_account_identifier": approps[0]["treasury_account"],
+            "fiscal_year": 2019,
+            "fiscal_period": 3,
+            "obligations_incurred_total_cpe": 1,
+        },
+        {
+            "treasury_account_identifier": approps[1]["treasury_account"],
+            "fiscal_year": 2019,
+            "fiscal_period": 4,
+            "obligations_incurred_total_cpe": 2,
+        },
+        {
+            "treasury_account_identifier": approps[0]["treasury_account"],
+            "fiscal_year": 2019,
+            "fiscal_period": 5,
+            "obligations_incurred_total_cpe": 3,
+        },
+        {
+            "treasury_account_identifier": approps[0]["treasury_account"],
+            "fiscal_year": 2020,
+            "fiscal_period": 3,
+            "obligations_incurred_total_cpe": 4,
+        },
+        {
+            "treasury_account_identifier": approps[1]["treasury_account"],
+            "fiscal_year": 2020,
+            "fiscal_period": 3,
+            "obligations_incurred_total_cpe": 5,
+        },
+        {
+            "treasury_account_identifier": approps[1]["treasury_account"],
+            "fiscal_year": 2020,
+            "fiscal_period": 3,
+            "obligations_incurred_total_cpe": 6,
+        },
+    ]
+    for gtas in gtas_rows:
+        mommy.make(
+            "references.GTASSF133Balances",
+            treasury_account_identifier=gtas["treasury_account_identifier"],
+            fiscal_year=gtas["fiscal_year"],
+            fiscal_period=gtas["fiscal_period"],
+            obligations_incurred_total_cpe=gtas["obligations_incurred_total_cpe"],
+        )
+
+
+def test_run_script(setup_test_data):

```

Comment:
Difficult for me to tell if this single test function captures enough edge cases. It looks like a single happy path

Reply:
Should I comment the cases? It includes results for a case that needed to be summed, one that is a single row (still summed, but...), and ones that should not be in the results.

Reply:
good to know